ddtrace/span.py

```diff
@@ -521,3 +521,11 @@ def __repr__(self):
             self.parent_id,
             self.name,
         )
+
+
+def _is_top_level(span):
+    # type: (Span) -> bool
+    """Return whether the span is a "top level" span."""
+    return (span._local_root is span) or (
+            span._parent is not None and span._parent.service != span.service and span.service is not None
```

Comment:
> and span.service is not None

is a new addition here. any reason why?

Reply:
Yeah it handles the implicit inheritance that occurs with service=None. If service=None then the service gets inherited.

So if the parent span has a service and the span's service is None then span should not be marked at top level.

Reply:
probably important that we check these in this order, since if the span is the local root, service=None is acceptable.

maybe we should leave a comment about ordering of these checks?

Reply:
I added testcases to ensure this functionality which should be a better enforcement than a comment.